File: solns/ruby/12.rb
def tau(n)
  return 1 if n == 1

  2 * (1..Math.sqrt(n)).count { |i| (n % i).zero? }
end

n = 3
i = 3
while tau(n) < 500
  n += i
  i += 1
end
result = n

puts result
raise Error unless result == 76_576_500

File: solns/haskell/12.hs
tau :: Int -> Int
tau 1 = 1
tau n = 2 * x
        where sqn = floor . sqrt . fromIntegral $ n
              x = length $ filter (\i -> mod n i == 0) [1..sqn]

find n i | tau n > 500 = n
find n i = find (n+i) (i+1)

result = find 3 3
main = print result
